File: module_main/src/main/java/com/kedacom/module_main/MainActivity.java
package com.kedacom.module_main;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;

import com.kedacom.module_common.constant.RouteConstants;
import com.kedacom.module_common.service.ILoginService;
import com.kedacom.module_common.service.ServiceFactory;
import com.kedacom.module_lib.base.common.BaseActivity;
import com.kedacom.module_lib.utils.LogUtil;
import com.kedacom.module_lib.utils.ToastUtil;
import com.kedacom.module_main.adapter.MainAdapter;

public class MainActivity extends BaseActivity {

    RecyclerView mRecyclerView;

    private SparseArray<String> sparseArray;
    private MainAdapter functionListAdapter;

    public static void newInstance(Context context) {
        //显示Intent
//        Intent intent = new Intent(context, FunctionListActivity.class);
//        context.startActivity(intent);

//        隐式Intent
        String mAction = context.getPackageName() + ".FunctionList";
        Intent intent = new Intent(mAction);
        context.startActivity(intent);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        LogUtil.myD("44444");

        setContentView(R.layout.activity_main);

        initData();


        ILoginService accountService = ServiceFactory.getInstance().getAccountService();

        if (accountService.isLogin()) {
            LogUtil.myD("用户已登录" + accountService.getAccountId());
        } else {
            ToastUtil.showShortToast(mActivity, "请先登录！");
        }
        mRecyclerView.setLayoutManager(new LinearLayoutManager(mActivity));
        mRecyclerView.setAdapter(functionListAdapter);
    }


    protected void initData() {
        sparseArray = new SparseArray<>();

        sparseArray.put(0, RouteConstants.LEARN_FUNCTION_LIST);
        sparseArray.put(1, RouteConstants.NEWS_ACTIVITY);

        functionListAdapter = new MainAdapter(mActivity, sparseArray);
    }
}

File: module_main/src/main/java/com/kedacom/module_main/MainActivity.kt
package com.kedacom.module_main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.util.Log
import com.alibaba.android.arouter.facade.annotation.Route
import com.kedacom.module_common.constant.RouteConstants
import com.kedacom.module_common.common.BaseActivity
import com.kedacom.module_main.adapter.MainAdapter
import com.kedacom.module_main.bean.ItemBean
import com.kedacom.module_main.mvp.view.MvpDemoActivity
import com.kedacom.module_main.mvvm.MvvmDemoActivity
import kotlinx.android.synthetic.main.activity_main.*

@Route(path = RouteConstants.MAIN_ACTIVITY)
class MainActivity : BaseActivity() {
    private lateinit var list: ArrayList<ItemBean>
    private lateinit var mainAdapter: MainAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("caowj", "44444")
        setContentView(R.layout.activity_main)
        initData()

        mainList.layoutManager = LinearLayoutManager(mActivity)
        mainList.adapter = mainAdapter
    }

    private fun initData() {
        list = ArrayList<ItemBean>()
        list.add(ItemBean("MvpDemo", MvpDemoActivity::class.java))
        list.add(ItemBean("MvvmDemo", MvvmDemoActivity::class.java))
        mainAdapter = MainAdapter(mActivity, list)
    }

    companion object {
        fun newInstance(context: Context) {
            context.startActivity(Intent(context, MainActivity::class.java))
        }
    }
}